fhevm/tee_interpreter.go

```diff
@@ -50,7 +50,12 @@ func doOp(
 	// Note that we do arithmetic operations on uint64, then we convert the
 	// result back to the FheUintType.
 	l := big.NewInt(0).SetBytes(lp.Value).Uint64()
-	r := big.NewInt(0).SetBytes(rp.Value).Uint64()
+	var r uint64
+	if !isScalar {
+		r = big.NewInt(0).SetBytes(rp.(*tee.TeePlaintext).Value).Uint64()
+	} else {
+		r = big.NewInt(0).SetBytes(rp.([]byte)).Uint64()
+	}
```

Comment:
Does zama have any tests for scalar? We should add tests to tee scalar ops too

Reply:
Zama don't have any tests for scalar.

Reply:
@amaury1093 Do you wanna I add tests to tee scalar ops too?